data/data/azure/vnet/common.tf

```diff
@@ -6,15 +6,15 @@ data "azurerm_subnet" "preexisting_master_subnet" {
 
   resource_group_name  = var.azure_network_resource_group_name
   virtual_network_name = var.azure_virtual_network
-  name                 = var.azure_control_plane_subnet
+  name                 = var.azure_control_plane_subnet[0]
```

Comment:
Shouldn't this use an index instead of 0? How do you setup more than a single subnet in install-config and pass it into terraform ?

Reply:
When bringing their own network, users were expected to provide only one subnet for control plane and one for compute. So I'm using `[0]` to keep the previous logic explicit here. I can change it to an index but the effect is the same given the `count` is either `0` or `1`.